### PR TITLE
Bug fix for the retry mechanism

### PR DESCRIPTION
**Pull request by bonomat**

### CODE REVIEW

File: src/actionExecutor.ts [hunk 1]
```diff
@@ -52,10 +55,9 @@ export class ActionExecutor {
     } else if (maxRetries <= 0) {
       result = Result.err(new Error("Maximum number of retries reached"));
     } else {
-      // It failed, try again in 30 s
-      setInterval(() => {
-        this.execute(action, maxRetries - 1);
-      }, 30000);
+      // It failed, try again in x milliseconds
+      await this.sleep(timeout);
+      return this.execute(action, maxRetries - 1);
```
Comment (D4nte) [inline src/actionExecutor.ts]:
```suggestion
      return this.execute(action, maxRetries - 1, timeout);
```

File: tests/actionExecutor.spec.ts [hunk 2]
```diff
@@ -120,6 +120,41 @@ describe("Action executor tests: ", () => {
 
     done();
   });
+
+  it("should retry action twice", async done => {
```
Comment (da-kami) [inline tests/actionExecutor.spec.ts]: 🙃 Great that there is a test for this now. Could we add another one that tests that we are not running into an endless loop of retries if the comit-node is actually down? (`it fails when retrying more often than max retries` or similar)

File: tests/actionExecutor.spec.ts [hunk 3]
```diff
@@ -120,6 +120,78 @@ describe("Action executor tests: ", () => {
...
+    nock("http://localhost:8000")
+      .get("/swaps/rfc003/")
+      .reply(200, swapsAcceptDeclineStub);
+    const scope = nock("http://localhost:8000")
+      .post("/swaps/rfc003/399e8ff5-9729-479e-aad8-49b03f8fc5d5/accept")
+      .replyWithError("Oh no, we ran out of bananas.")
+      .post("/swaps/rfc003/399e8ff5-9729-479e-aad8-49b03f8fc5d5/accept")
+      .replyWithError("You're telling me still no bananas?")
+      .post("/swaps/rfc003/399e8ff5-9729-479e-aad8-49b03f8fc5d5/accept")
+      .replyWithError("🙈🙉🙊");
```
Comment (da-kami) [inline tests/actionExecutor.spec.ts]: ❤️ beautiful monkey faces in the test code

File: tests/actionExecutor.spec.ts [hunk 4]
```diff
@@ -120,6 +120,78 @@ describe("Action executor tests: ", () => {
...
+      action => action.name === "accept"
+    ) as Action;
+
+    const actionResponse = await actionExecutor.execute(acceptAction, 2, 2);
+
+    expect(actionResponse).toStrictEqual(
+      Result.err(new Error("Maximum number of retries reached"))
+    );
+
+    // check the mock was invoked indeed 3 times
```
Comment (da-kami) [inline tests/actionExecutor.spec.ts]:
```suggestion
    // check the mock finished execution as expected
```
It does not really check the exact number of times here (but they are implied).

**Pull request merged**